airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination/buffered_stream_consumer/BufferedStreamConsumer.java

```diff
@@ -101,19 +105,35 @@ public BufferedStreamConsumer(final Consumer<AirbyteMessage> outputRecordCollect
                                 final ConfiguredAirbyteCatalog catalog,
                                 final CheckedFunction<JsonNode, Boolean, Exception> isValidRecord,
                                 final long maxQueueSizeInBytes) {
+    this(outputRecordCollector, onStart, recordWriter, null, null,
+        onClose, catalog, isValidRecord, maxQueueSizeInBytes);
+  }
+
+  public BufferedStreamConsumer(final Consumer<AirbyteMessage> outputRecordCollector,
+                                final VoidCallable onStart,
+                                final RecordWriter recordWriter,
+                                final RecordWriter removeRecordWriter,
+                                final GetFileRecordWriter getFileRecordWriter,
+                                final CheckedConsumer<Boolean, Exception> onClose,
+                                final ConfiguredAirbyteCatalog catalog,
+                                final CheckedFunction<JsonNode, Boolean, Exception> isValidRecord,
+                                final long maxQueueSizeInBytes) {
     this.outputRecordCollector = outputRecordCollector;
     this.maxQueueSizeInBytes = maxQueueSizeInBytes;
     this.hasStarted = false;
     this.hasClosed = false;
     this.onStart = onStart;
     this.recordWriter = recordWriter;
+    this.removeRecordWriter = removeRecordWriter;
+    this.getFileRecordWriter = getFileRecordWriter;
     this.onClose = onClose;
     this.catalog = catalog;
     this.streamNames = AirbyteStreamNameNamespacePair.fromConfiguredCatalog(catalog);
     this.isValidRecord = isValidRecord;
     this.buffer = new ArrayList<>(10_000);
     this.bufferSizeInBytes = 0;
     this.pairToIgnoredRecordCount = new HashMap<>();
+    this.fileName = null;
```

Comment:
Please, avoid null for String value, instead use empty string. And for checking empty value we can use StringUtils.isEmpty() or StringUtils.isBlank()

Reply:
updated it